snapcraft/plugins/jhbuild.py

```diff
@@ -0,0 +1,1035 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2016 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The JHBuild plugin is used for GNOME-based parts.
+
+JHBuild is a tool used to build components of the GNOME ecosystem from version
+control. See the JHBuild manual for more details:
+https://developer.gnome.org/jhbuild/stable/.
+
+The plugin can be customized with the following keys:
+
+  - modules (required): a list of modules to include in the part, modules that
+                        should be skipped should be prefixed with '-'
+  - moduleset: the module set JHBuild should use (default: gnome-world)
+  - snap-name (required): the name of the snap this part is included in
+  - snap-revision: the revision of the snap this part is included in
+                   (default: current)
+  - container-name: the name of the LXC container used for building
+                    (default: snapcraft-<snap name>-<part name>)
+  - distribution: the distribution the LXC container is based off of
+                  (default: host distribution)
+  - release: the release of the distribution the LXC container is based off of
+             (default: host release)
+  - architecture: the architecture of the LXC container
+                  (default: host architecture)
```

Comment:
I like that you can select the image you want, but I think this should be more general, not just for this plugin.
We had https://bugs.launchpad.net/snapcraft/+bug/1597952 for this.